src/chart-addons/stacked-legend.js

```diff
@@ -153,7 +153,8 @@ export function handleLegendLock({ locked, index = 0 }) {
   let redraw = false
   if (typeof color.scale === "object") { // this if or raster-layer-heatmap-mixin.js
     if (color.legend.locked) {
-      layer.setState(setColorScaleDomain_v2(layer.colorDomain()))
+      const colorDomain = color.scale.domain ? color.scale.domain : layer.colorDomain()
```

Comment:
the value of `layer.colorDomain()` could be `null` here, does that matter?

Reply:
also is passing `color.scale.domain` okay here? It looks like it's passing an array by reference, might be better to clone the array before passing? maybe it doesn't matter.

Reply:
either color.scale.domain or layer.colorDomain() expects an value. In initial null cases, domain should be populated with auto scale min/max.

Reply:
In this case even we pass the array by reference, we not directly modifying the array.